tests/quickcheck.rs

```diff
@@ -0,0 +1,283 @@
+fn quickcheck_signed_ge_le_eq_mut_exclusive(a: BigInt, b: BigInt) -> bool {
+    let gt_lt_eq = vec![a > b, a < b, a == b];
+    gt_lt_eq
+        .iter()
+        .fold(0, |acc, e| if *e { acc + 1 } else { acc })
+        == 1
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_sub_primitive(a: u128, b: u128) -> bool {
+    if b < a {
+        BigUint::from(a - b) == BigUint::from(a) - BigUint::from(b)
+    } else {
+        BigUint::from(b - a) == BigUint::from(b) - BigUint::from(a)
+    }
+}
+
+#[quickcheck]
+fn quickcheck_signed_sub_primitive(a: i128, b: i128) -> bool {
+    if b < a {
+        BigInt::from(a - b) == BigInt::from(a) - BigInt::from(b)
+    } else {
+        BigInt::from(b - a) == BigInt::from(b) - BigInt::from(a)
+    }
+}
+
+#[quickcheck]
+/// Tests correctness of subtraction assuming addition is correct
+fn quickcheck_unsigned_sub(a: BigUint, b: BigUint) -> bool {
+    if b < a {
+        a.clone() - b.clone() + b == a
+    } else {
+        b.clone() - a.clone() + a == b
+    }
+}
+
+#[quickcheck]
+/// Tests correctness of subtraction assuming addition is correct
+fn quickcheck_signed_sub(a: BigInt, b: BigInt) -> bool {
+    if b < a {
+        a.clone() - b.clone() + b == a
+    } else {
+        b.clone() - a.clone() + a == b
+    }
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_div_primitive(a: u128, b: u128) -> TestResult {
+    if b == 0 {
+        TestResult::discard()
+    } else {
+        TestResult::from_bool(BigUint::from(a / b) == BigUint::from(a) / BigUint::from(b))
+    }
+}
+
+#[quickcheck]
+fn quickcheck_signed_div_primitive(a: i128, b: i128) -> TestResult {
+    if b == 0 {
+        TestResult::discard()
+    } else {
+        TestResult::from_bool(BigInt::from(a / b) == BigInt::from(a) / BigInt::from(b))
+    }
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_pow_zero(a: BigUint) -> bool {
+    a.pow(0_u32) == BigUint::one()
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_pow_one(a: BigUint) -> bool {
+    a.pow(1_u32) == a
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_sqrt(a: BigUint) -> bool {
+    (a.clone() * a.clone()).sqrt() == a
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_cbrt(a: BigUint) -> bool {
+    (a.clone() * a.clone() * a.clone()).cbrt() == a
+}
+
+#[quickcheck]
+fn quickcheck_signed_cbrt(a: BigInt) -> bool {
+    (a.clone() * a.clone() * a.clone()).cbrt() == a
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_conversion(a: BigUint) -> bool {
+    let mut success = true;
+    for radix in 2..=36 {
+        let string = a.to_str_radix(radix);
+        success &= a == BigUint::from_str_radix(&string, radix).unwrap()
+    }
+    success
+}
+
+#[quickcheck]
+fn quickcheck_signed_conversion(a: BigInt) -> bool {
+    let mut success = true;
+    for radix in 2..=36 {
+        let string = a.to_str_radix(radix);
+        success &= a == BigInt::from_str_radix(&string, radix).unwrap()
+    }
+    success
+}
+
+#[quickcheck]
+fn quickcheck_unsigned_modpow(a: BigUint, exp: u8, modulus: BigUint) -> TestResult {
```

Comment:
Out of all the tests, this one actually took the longest time to run (~7s), and it isn't very useful, so I'm removing it.